### ISSUE TITLE
Review server.start args and environment variable handling

### ISSUE THREAD

**Issue by JedWatson** [keystonejs/keystone-5#942]
Reviewing #941 I noticed that our `server.start` method takes the MongoDB connection URL as its first argument.

This struck me as really obscure, I never would have expected that. I'd like to see how we can make it clearer - we use a lot of argument destructuring in Keystone so options are self-documenting, and this really breaks that clarity.

In addition, in Keystone 4 we detected common environment variables like `PORT` and `MONGODB_URI`, I think we're not doing that yet in Keystone 5. Unless there's some new opinion about not doing that, I think it's a nice touch that removes boilerplate, and we should start doing it.

Documented here: https://keystonejs.com/documentation/configuration

Specifically:

* `env: process.env.NODE_ENV || 'development'`
* `port: process.env.PORT || 3000`
* `host: process.env.IP || '127.0.0.1'`
* `mongo: process.env.MONGO_URI || 'mongodb://localhost/{db-name}'` and also [all of these](https://github.com/keystonejs/keystone/blob/master/lib/core/initDatabaseConfig.js) (we can probably seriously cull this list, but should support the heroku default at minimum)

**Comment by JedWatson** [comment]
Also probably related to #926

**Comment by w01fgang** [comment]
Yes, I miss them too. Having env variables would be great.
I just played with the demo projects. I was looking for the option `mongo` or similar.
Also, I found that mongoose errors are lost because of unhandled rejection in `packages/utils/src/index.js`. After adding `.catch` after `.then` I found that the database name is taken from the app name even if I explicitly pass it with the mongo connection string to `.start` method.
